packages/controller-utils/src/siwe.ts

```diff
@@ -0,0 +1,53 @@
+import { ParsedMessage } from '@spruceid/siwe-parser';
+import { bytesToString, hexToBytes } from '@metamask/utils';
+
+const msgHexToText = (hex: string): string => {
+  try {
+    const bytes = hexToBytes(hex);
+    return bytes.length === 32 ? hex : bytesToString(bytes);
+  } catch (e) {
+    return hex;
+  }
+};
+
+/**
+ * A locally defined object used to provide data to identify a Sign-In With Ethereum (SIWE)(EIP-4361) message and provide the parsed message
+ *
+ * @typedef localSIWEObject
+ * @param {boolean} isSIWEMessage - Does the intercepted message conform to the SIWE specification?
+ * @param {ParsedMessage} parsedMessage - The data parsed out of the message
+ */
+export interface SIWEMessage {
```

Comment:
Suggestion: We could improve/further restrict this type by making it a union of the "success" and "failure" responses. That would prevent situations like `{ isSIWEMessage: false, parsedMessage: ParsedMessage }` and `{ isSWIEMessage: true, parsedMessage: null }` at the type-level, which should reduce the validation needed to use this function from a TypeScript project.

Maybe something like `SWIEMessage = { isSIWEMEssage: true, parsedMessage: ParsedMessage } || { isSIWEMessage: false, parsedMessage: null }`

Reply:
I see what you mean @Gudahtt, that's a great suggestion, fixed.